pkg/control/sidecarcontrol/util.go

```diff
@@ -253,11 +254,81 @@ func UpdatePodSidecarSetHash(pod *corev1.Pod, sidecarSet *appsv1alpha1.SidecarSe
 		SidecarSetName:               sidecarSet.Name,
 		SidecarList:                  sidecarList.List(),
 		SidecarSetControllerRevision: sidecarSet.Status.LatestRevision,
+		State:                        "Updating",
```

Comment:
Maybe should define the `State` using the `Enum` type or make it a const.

Reply:
done